src/main/java/org/asteriskjava/manager/util/EventAttributesHelper.java

```diff
@@ -81,12 +82,25 @@ public static void setAttributes(Object target, Map<String, Object> attributes,
             // it seems silly to warn if it's a user event -- maybe it was
             // intentional
             if (setter == null && !(target instanceof UserEvent) && !target.getClass().equals(ManagerResponse.class)) {
-                logger.warn("Unable to set property '" + entry.getKey() + "' to '" + entry.getValue() + "' on "
-                    + target.getClass().getName()
-                    + ": no setter. Please report at https://github.com/asterisk-java/asterisk-java/issues");
 
-                for (Map.Entry<String, Object> entry2 : attributes.entrySet()) {
-                    logger.debug("Key: " + entry2.getKey() + " Value: " + entry2.getValue());
+                //CDR has dynamic properties
+                if (target instanceof CdrEvent) {
+                    try {
+                        Method addproperty= CdrEvent.class.getMethod("addDynamicProperties", String.class, String.class);
```

Comment:
I don't fully understand this. Why we need to do this manually?

Reply:
as today CDRevent is not adding the dynamic CDR variables, invokers use setters for existing fields, but this could be anything you add in dialplan

Reply:
in this case the props are empty on porpoise for testing but this is what I mean, you'll see guid, direction, and others
<img width="1680" alt="Screenshot 2023-05-25 at 14 30 40" src="https://github.com/asterisk-java/asterisk-java/assets/984668/0250e6b8-b55c-478e-8411-a6d2e6b0135b">